ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -288,6 +292,10 @@ applyChainTick ::
   -> Ticked1 l mk
 applyChainTick = lrResult ..: applyChainTickLedgerResult
 
+
+class (Ord k, Serialise k, Serialise v) => LedgerConstraint k v
```

Comment:
Before we merge, we should avoid using `Serialise` for this.

My suggestion is a newtype (like my `LmdbBox` type from the db-analyser patch branch you saw) that's wrapped and unwrapped immediately before passing/after receiving data from `lmdb-simple`. That newtype would have a `Serialise` instance (since `lmdb-simple` demands it) that directly defers to our preferred class (which can start as `ToCBOR` and `FromCBOR`, but maybe should ultimately be our `EncodeDisk` class? I'm unsure).